src/main/java/org/broadinstitute/hellbender/tools/longreads/graph/AlignedBaseGraphCollection.java

```diff
@@ -0,0 +1,583 @@
+                    if ( vertex.isSequenceEqual(graphVertex.getSequence()) ) {
+                        // Setting vertex here will cause lastVertex to be set later.
+                        vertex = graphVertex;
+                        vertexIsInGraph = true;
+
+                        // If we added our last node, we need to make a link from the newly-added last node to
+                        // the current graph node.
+                        if ( lastNodeWasAddedToGraph ) {
+                            // This if statement can probably be removed:
+                            if (lastVertex.equals(vertex)) {
+                                logger.error("Equal Vertices detected: " + vertex.toString() + " @ " + vertex.getPos().toString());
+                                throw new GATKException("STILL BUSTED, BUSTER!");
+                            }
+                            graph.addEdge(lastVertex, vertex);
+                        }
+                        lastNodeWasAddedToGraph = false;
+                        break;
+                    }
+                }
+            }
+
+            // We must add in the new vertex and its edges:
+            if ( !vertexIsInGraph ) {
+                // Add our vertex to the graph:
+                graph.addVertex(vertex);
+
+                // Add the vertex to our positional data store:
+                if ( positionVertexMap.containsKey(vertex.getPos()) ) {
+                    positionVertexMap.get(vertex.getPos()).add(vertex);
+                }
+                else {
+                    positionVertexMap.put(
+                            vertex.getPos(),
+                            new HashSet<>(Collections.singletonList(vertex))
+                    );
+                }
+
+                // Add an edge from the previous position if we must:
+                if (lastVertex != null) {
+                    graph.addEdge(lastVertex, vertex);
+                }
+                lastNodeWasAddedToGraph = true;
+            }
+
+            lastVertex = vertex;
+        }
+    }
+
+    /**
+     * Create a linear sequence of node objects based on aligned sequence information.
+     * @param bases Bases to be added in the order in which they appear in the read.
+     * @param cigar {@link List<CigarElement>} containing alignment information for each base in {@code bases}.
+     * @param contig The contig to which the given base sequence is aligned.
+     * @param startPos The start position of the first aligned base in the given base sequence.
+     * @param readName The name of the parent read from which this aligned sequence information is derived.
+     * @return A {@link Queue<AlignedBaseVertex>} representing a linear graph of bases in a given read.
+     */
+    private Queue<AlignedBaseVertex> createAlignedNodes(final byte[] bases, final List<CigarElement> cigar, final String contig, final int startPos, final String readName) {
```

Comment:
As per the discussion we had about this. If you are concerned about performance/memory usage I think this code would be a place to optimize. This method creates an `AlignedBaseVertex` object (and its associated fields and object fields) for every base of every read (a calculation that can be approximated as average coverage x genome size; or 5-30 x 3,000,000,000 in your case). And then throwing the majority of these objects away downstream. It might be worth using the existing vertexes in the graph (and your map of position -> vertexes) as a cache here rather than creating a long string of aligned nodes and realigning them. 
Even faster might be to follow what we do for M2/HC local assembly and rather than creating an object in every case, simply do this operation directly onto the graph first checking if the edge you want exists and if it does stopping and trying the next base. You can look at `AbstractReadThreadingGraph.extendChainByOne()` for an example of how to do this. I suspect equivalent behavior to how that code works would save a lot of time and memory to what you are doing now.